### PR TITLE
Incremental staged sync for pruning

### PR DESCRIPTION
**Pull request by vorot93**

### CODE REVIEW

File: eth/stagedsync/stage_headers_new.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ type HeadersCfg struct {
 	penalize          func(context.Context, []headerdownload.PenaltyItem)
 	wakeUpChan        chan struct{}
 	batchSize         datasize.ByteSize
+	increment         *uint64
```
Comment (AlexeyAkhunov) [inline eth/stagedsync/stage_headers_new.go:29]: Why is this a pointer?

File: eth/stagedsync/stage_headers_new.go [hunk 2]
```diff
@@ -37,6 +38,7 @@ func StageHeadersCfg(
 	penalize func(context.Context, []headerdownload.PenaltyItem),
 	wakeUpChan chan struct{},
 	batchSize datasize.ByteSize,
+	increment *uint64,
```
Comment (AlexeyAkhunov) [inline eth/stagedsync/stage_headers_new.go:41]: Why is this a pointer?

File: eth/stagedsync/stage_headers_new.go (hunk 1 above)
Comment (vorot93) [reply eth/stagedsync/stage_headers_new.go]: done

**Review by AlexeyAkhunov — APPROVED** [review]

File: turbo/cli/flags.go [hunk 3]
```diff
@@ -55,6 +55,11 @@ var (
 		Usage: "enable experimental downloader v2",
 	}
 
+	PruningFlag = cli.BoolFlag{
+		Name:  "prune",
+		Usage: "Enable pruning ancient data",
```
Comment (AskAlexSharov) [inline turbo/cli/flags.go:60]: Add comment that this feature experimental yet. If pruning it’s storage mode, why it’s not part of —storage-mode flag?

Comment (vorot93) [reply turbo/cli/flags.go:60]: Mostly for reasons of aesthetics and discoverability. Pruning is not _just_ another optional data item - it's a substantially different sync mode. Also, this flag will be set _much more often_ than other storage mode toggles.

I only put pruning inside storage mode to reuse the code around checking mode compatibility.

**Pull request merged**